libs/community/langchain_community/chat_models/deepinfra.py

```diff
@@ -417,6 +421,27 @@ def _headers(self) -> Dict:
     def _body(self, kwargs: Any) -> Dict:
         return kwargs
 
+    def bind_tools(
```

Comment:
@ccurme could you recommend any standard tests that can be used to verify this implementation?

Reply:
Thanks for the comment @eyurtsev, let me check it out too.

Reply:
@ovuruska we have some standard unit tests in some of the partner packages -- i'd check langchain_openai / langchain_anthropic to see if there's something there that could be help run through all the relevant tests for tool usage

Reply:
@ovuruska thanks for this. In addition to your test that covers the tool binding, I'd suggest an integration test that invokes the model and verifies that a tool was called. Example here: https://github.com/langchain-ai/langchain/blob/8203c1ff874e66550b5656ab054c2b946df2ac00/libs/partners/openai/tests/integration_tests/chat_models/test_base.py#L537

---

Separately, It looks like tool calls are ending up in `additional_kwargs` here. You might consider formatting them into the [AIMessage.tool_calls](https://api.python.langchain.com/en/latest/messages/langchain_core.messages.ai.AIMessage.html#langchain_core.messages.ai.AIMessage.tool_calls) attribute in addition (or instead). This would make it easier to support DeepInfra models in langchain and langgraph agents and other applications. IMO this is a non-blocking concern but would have some benefit.

Example here:
https://github.com/langchain-ai/langchain/blob/8203c1ff874e66550b5656ab054c2b946df2ac00/libs/partners/openai/langchain_openai/chat_models/base.py#L109-L130